main.py

```diff
@@ -59,6 +58,31 @@ def on_keypress(bit_index):
   clock.tick(60)
   pygame.display.flip()
 
+  if not game.state_of_game:
+    draw_layout(game, binary_boxes, binary_bar_preview, score_display)
+    for event in pygame.event.get():
+      match event.type:
+        case pygame.QUIT:
+          pygame.quit()
+          exit()
+        case pygame.KEYDOWN:
+          if event.key == pygame.K_SPACE:
+            game.state_of_game = True
+            draw_layout(game, binary_boxes, binary_bar_preview, score_display)
+            # resetting values to default
+            time_since_enemy_spawn = time()
+            time_between_spawns = 5
+            mergers = {}
+            shot_missiles = {}
+            game.alive_enemies = []
+            game.score = 0
+            score_display.text_content = "0"
+            score_display.draw_display()
+            for box in binary_boxes:
+              if box.current_bit: 
+                box.flip_bit()
```

Comment:
This resetting might benefit from being moved to a method on the game object to avoid bloating the main game loop.
Would also provide the label in the method name that you've used a comment to satisfy.